aggregate/aggregate

```diff
@@ -0,0 +1,189 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import pandas as pd
+import re
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Aggregate results from mentha2pdb, string2pdb, pdbminer, and pdbminer_complexes.")
+
+    parser.add_argument(
+        "-m", 
+        required=True,
+        help="Path to the mentha2pdb output csv file.")
+
+    parser.add_argument(
+        "-s",
+        required=True,
+        help="Path to the string2pdb output csv file.")
+
+    args = parser.parse_args()
+
+    # Load mentha2pdb output:
+    mentha_df = pd.read_csv(args.m)
+
+    # Load string2pdb output:
+    string_df = pd.read_csv(args.s)
+
+    ## MENTHA2PDB OUTPUT PRE-PROCESSING: ##
+    
+    # Replace 'na' values with "":
+    mentha_df.replace('na', "", inplace=True)
+
+    # Keep only the required columns
+    mentha_df = mentha_df[[
+        "target uniprot id", 
+        "target uniprot gene", 
+        "interactor uniprot id", 
+        "interactor uniprot gene", 
+        "mentha score", 
+        "PDB id", 
+        "pDockQ HuMap", 
+        "pDockQ HuRI"
+    ]]
+
+    mentha_df.rename(columns={
+        "target uniprot gene": "Target_protein",
+        "target uniprot id": "Target_Uniprot_AC",
+        "interactor uniprot gene": "Interactor",
+        "interactor uniprot id": "Interactor_UniProt_AC",
+        "mentha score": "Mentha_score"
+    }, inplace=True)
+
+    # Simplify gene names by keeping only the part before the first space or '{' :
+    mentha_df["Target_protein"] = mentha_df["Target_protein"].str.replace(r"[ {].*", "", regex=True)
+    mentha_df["Interactor"] = mentha_df["Interactor"].str.replace(r"[ {].*", "", regex=True)
+
+    # Initialize the "Structure" column with the PDB id values:
+    mentha_df["Structure"] = mentha_df["PDB id"].fillna("")
+
+    # Add "AF_Huri_HuMAP" to "Structure" if HuMap or HuRI values are not "":
+    mentha_df["Structure"] = mentha_df.apply(
+        lambda row: ",".join(filter(None, [row["Structure"], "AF_Huri_HuMAP"])) \
+        if row["pDockQ HuMap"] != "" or row["pDockQ HuRI"] != "" else row["Structure"],
+        axis=1
+    )
+
+    # Drop columns "PDB id", "pDockQ HuMap", and "pDockQ HuRI":
+    mentha_df.drop(columns=["PDB id", "pDockQ HuMap", "pDockQ HuRI"], inplace=True)
+
+    # Group by all columns except "Structure" and aggregate "Structure" as comma-separated strings
+    mentha_df = mentha_df.groupby([
+        "Target_Uniprot_AC", 
+        "Target_protein", 
+        "Interactor_UniProt_AC", 
+        "Interactor", 
+        "Mentha_score"
+    ], as_index=False).agg({"Structure": lambda x: ",".join(sorted(set(filter(None, x))))})
```

Comment:
using a separator that isn't comma would be better here, because comma is the separator of csv columns already